tests/test_decompose.py

```diff
@@ -254,7 +254,7 @@ def test_nn_filter_badselfsim(data, x, y, sparse):
 
     srand()
     # Build a recurrence matrix, just for testing purposes
-    rec = np.random.randn(x, y)
+    rec: np.ndarray | scipy.sparse.csr_matrix = np.random.randn(x, y)
```

Comment:
Ah this gets tricky.  I recently merged a huge overhaul of the test suite into the 1.0 dev branch (not yet merged into main), which among other things, involved a total rewrite of RNG usage in the test suite.

It's a little orthogonal to the changes in this PR, but we're headed for a merge conflict...

Reply:
I personally don't mind dealing with merge conflicts. So if that will happen, then I'd be happy to help out.